Lib/test/test_capi/test_set.py

```diff
@@ -0,0 +1,175 @@
+import unittest
+
+from test.support import import_helper
+
+# Skip this test if the _testcapi module isn't available.
+_testcapi = import_helper.import_module('_testcapi')
+
+class set_child(set):
+    pass
+
+class frozenset_child(frozenset):
+    pass
+
+
+class TestSetCAPI(unittest.TestCase):
+    def assertImmutable(self, action, *args):
+        self.assertRaises(SystemError, action, frozenset(), *args)
```

Comment:
SystemError? That's a surprising error. Usually, it's used when the C API is misused, like passing NULL or the wrong type. frozenset is a "wrong type" here?

Reply:
Yes, immutable frozenset is a wrong type for mutation-based functions :)